src/tools/storage-key-recipe-resolver.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Runtime} from '../runtime/runtime.js';
+import {Manifest} from '../runtime/manifest.js';
+import {Loader} from '../platform/loader-web.js';
+import {IsValidOptions, Recipe, RecipeComponent} from '../runtime/recipe/recipe.js';
+import {ramDiskStorageKeyPrefixForTest} from '../runtime/testing/handle-for-test.js';
+import {Arc} from '../runtime/arc.js';
+import {RecipeResolver} from '../runtime/recipe/recipe-resolver.js';
+import {CapabilitiesResolver} from '../runtime/capabilities-resolver.js';
+import {Store} from '../runtime/storageNG/store.js';
+import {Exists} from '../runtime/storageNG/drivers/driver.js';
+
+/**
+ * Responsible for resolving recipes with storage keys.
+ */
+export class StorageKeyRecipeResolver {
+  private readonly runtime: Runtime;
+
+  constructor(context: Manifest) {
+    const loader = new Loader();
+    this.runtime = new Runtime({loader, context});
+  }
+
+  /**
+   * Produces resolved recipes with storage keys.
+   *
+   * TODO(alxr): Apply to long-running recipes appropriately.
+   * @throws Error if recipe fails to resolve on first or second pass.
+   * @yields Resolved recipes with storage keys
+   */
+  async* resolve(): AsyncGenerator<Recipe> {
```

Comment:
Is there an actual need for a generator? Will you sometimes be consuming only some of the outputs? Is it important to consume multiple streams and save memory by not materialising them?

I don't think so. I think we want to get the list of resolved recipes. Using generators where they are not adding value is making code more complex, don't do it unless there is a need.